main.inc.php

```diff
@@ -0,0 +1,293 @@
+
+if (isset($_GET['category'])) {
+	$category = intval($_GET['category']);
+	$_SESSION[$settings['session_prefix'].'usersettings']['category'] = $category;
+	$_SESSION[$settings['session_prefix'].'usersettings']['page'] = 1;
+}
+
+if (isset($category_ids) && isset($_SESSION[$settings['session_prefix'].'usersettings']['category_selection'])) {
+	$category_selection = filter_category_selection($_SESSION[$settings['session_prefix'].'usersettings']['category_selection'], $category_ids);
+	if (!empty($category_selection)) $category_selection_query = implode(', ', $category_selection);
+}
+
+// count spam:
+$count_spam_sql = 
+	"SELECT COUNT(*) FROM " . $db_settings['forum_table'] . " INNER JOIN (SELECT eid FROM " . $db_settings['akismet_rating_table'] . " WHERE " . $db_settings['akismet_rating_table'] . ".spam = 1 UNION SELECT eid FROM " . $db_settings['b8_rating_table'] . "	WHERE " . $db_settings['b8_rating_table'] . ".spam = 1) spam_list ON spam_list.eid = " . $db_settings['forum_table'] . ".id";
+$count_spam_result = mysqli_query($connid, $count_spam_sql);
+list($total_spam) = mysqli_fetch_row($count_spam_result);
+mysqli_free_result($count_spam_result);
+
+// show spam?  NOTE: variables are used in several php files, i.e. index.inc.php, thread.inc.php, entry.inc.php
+$show_spam = false;
+$spam_sql_and = " AND spam_list.id IS NULL";
+
+if ($total_spam > 0 && isset($_SESSION[$settings['session_prefix'].'usersettings']['show_spam'])) {
+	$show_spam = true;
+	$spam_sql_and = " AND spam_list.id IS NOT NULL";
+}
+
+// count postings, threads, users and users online:
+$total_threads_postings_body = 
+	"SELECT COUNT(DISTINCT " . $db_settings['forum_table'] . ".id) FROM " . $db_settings['forum_table'] . " LEFT JOIN (SELECT ".$db_settings['forum_table'].".id, ".$db_settings['forum_table'].".tid FROM ".$db_settings['forum_table']." INNER JOIN " . $db_settings['akismet_rating_table'] . " ON ".$db_settings['forum_table'].".id = " . $db_settings['akismet_rating_table'] . ".eid WHERE " . $db_settings['akismet_rating_table'] . ".spam = 1 UNION SELECT ".$db_settings['forum_table'].".id, ".$db_settings['forum_table'].".tid FROM ".$db_settings['forum_table']." INNER JOIN " . $db_settings['b8_rating_table'] . " ON ".$db_settings['forum_table'].".id = " . $db_settings['b8_rating_table'] . ".eid WHERE " . $db_settings['b8_rating_table'] . ".spam = 1)";
+$threads_check_sql = " AND pid = 0";
+if ($show_spam) {
+	$spam_totals =  " AS spam_list ON spam_list.tid = " . $db_settings['forum_table'] . ".tid WHERE spam_list.id IS NOT NULL";
+} else {
+	$spam_totals = " AS spam_list ON spam_list.id = " . $db_settings['forum_table'] . ".id WHERE spam_list.id IS NULL";
+}
+if ($categories == false) {
+	$total_postings_category = "";		
+} else {
+	// there are categories
+	$total_postings_category = " AND category IN (" . $category_ids_query . ")";		
+}
+$total_threads_sql = $total_threads_postings_body . $spam_totals . $total_postings_category . $threads_check_sql;
```

Comment:
I got an error at this lines, i.e.

> [18-Apr-2024 19:19:12 UTC] PHP Fatal error:  Uncaught TypeError: mysqli_free_result(): Argument #1 ($result) must be of type mysqli_result, string given in /forum/includes/main.inc.php:257
> Stack trace:
> #0 /forum/includes/main.inc.php(257): mysqli_free_result('220')
> #1 /forum/index.php(36): include('/forum/...')
> #2 {main}
>   thrown in /forum/includes/main.inc.php on line 257

Suggestion (just using a different variable):
```
$total_threads_sql = $total_threads_postings_body . $spam_totals . $total_postings_category . $threads_check_sql;
$count_result = mysqli_query($connid, $total_threads_sql);
list($total_threads) = mysqli_fetch_row($count_result);
mysqli_free_result($count_result);

$total_postings_sql = $total_threads_postings_body . $spam_totals . $total_postings_category;
$count_result = mysqli_query($connid, $total_postings_sql);
list($total_postings) = mysqli_fetch_row($count_result);
mysqli_free_result($count_result);
```

Reply:
Interesting that it worked fine on our site, but I see the issue. Let me fix and commit.

Reply:
My PHP version was 8.2 (I now check version 8.3). Do we use the same version? In any case, it is not a show stopper and does not overshadow your overall effort! So, every thing is fine and no further issue occurs so far. ;-)

Reply:
Ahh, I just checked.  We're only running php 7.4.

Reply:
Maybe this is the reason (but I'm not an expert.)

Reply:
That was my guess too, and the same solution.  It is now committed.

Reply:
Ahh, now I remember why we’re on php 7.4! MyLittleForum 2.4.x is not compatible with php 8.x, so we have to stay on php 7.x until upgrade time.